### PR TITLE
ER-1122: Lead on news

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/ER-1122

- ER-1122: Added lead on news
- ER-1122: Added migration of news content

See https://github.com/eReolen/base/pull/166 for related design changes.

### CODE REVIEW

File: features/breol_news/breol_news.features.field_instance.inc [hunk 1]
```diff
@@ -333,9 +385,10 @@ function breol_news_field_default_field_instances() {
 
   // Translatables
   // Included for use with string extractors like potx.
+  t('Brødtekst');
   t('Cover Image');
-  t('Indhold');
   t('Karouseller');
+  t('Manchet');
```
Comment (cableman) [inline features/breol_news/breol_news.features.field_instance.inc:391]: Should both be named the same ('Brødtekst' vs 'Manchet') or?

File: features/breol_news/breol_news.install [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Move news content from body to lead.
+ */
+function breol_news_update_7101() {
+  $nodes = node_load_multiple([], ['type' => 'breol_news']);
+  foreach ($nodes as $node) {
+    $node->field_lead = $node->body;
+    $node->body = [];
+    node_save($node);
+  }
+
```
Comment (cableman) [inline features/breol_news/breol_news.install:18]: Empty line

File: features/breol_news/breol_news.install [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * @file
+ * Install, update and uninstall functions for the breol_news module.
+ */
+
+/**
+ * Move news content from body to lead.
+ */
+function breol_news_update_7101() {
+  $nodes = node_load_multiple([], ['type' => 'breol_news']);
```
Comment (cableman) [inline features/breol_news/breol_news.install:12]: How many items are we migration here? Should the batch API have been used and what about the format of the content moved?

Screenshot 2021-04-13 at 15 13 44

**Review by cableman — CHANGES_REQUESTED** [review]

File: features/breol_news/breol_news.features.field_instance.inc (hunk 1 above)
Comment (rimi-itk) [reply features/breol_news/breol_news.features.field_instance.inc:391]: No, one should be named “Manchet“ and one should be named “Brødtekst”.

File: features/breol_news/breol_news.install (hunk 3 above)
Comment (rimi-itk) [reply features/breol_news/breol_news.install:12]: We're talking about around 500 nodes. The (text) format on the old and the new field is the same.

**Review by cableman — APPROVED** [review]

File: features/breol_news/breol_news.install [hunk 4]
```diff
@@ -8,11 +8,42 @@
...
-    $node->field_lead = $node->body;
-    $node->body = [];
-    node_save($node);
+function breol_news_update_7101(&$sandbox) {
+  $items_per_batch = 50;
+
+  // Initial setup.
+  if (!isset($sandbox['processed'])) {
+    // Load the nids to update.
+    $nids = db_query('SELECT nid FROM {node} WHERE type = :type', ['type' => 'breol_news'])->fetchCol();
```
Comment (cableman) [inline features/breol_news/breol_news.install:17]: Maybe an where to limit it to nodes already moved (ala `field_body != ‘’`) so if the database dies the process can be continued.

**Review by cableman — CHANGES_REQUESTED** [review]

**Review by cableman — APPROVED** [review]

**Review by cableman — APPROVED** [review]

**Pull request merged**